### PR TITLE
feat: data model and dashboards for ern-cranio

### PR DESCRIPTION
**Pull request by davidruvolo51**
This request introduces a new Vue application for ERN CRANIO. This dashboard contains public and private pages that need to be customised according to user metadata. Specifically, the pages will be rendered accordingly to the user's permission level and affiliation with one healthcare provider (i.e., institution).

## Deliverables

- [x] Data model with test data
- [x] Website with public and private pages that render according to affiliation with an institution.

## Initial Tasks

- [x] Initial data model for CRANIO: convert emx1 model to emx2
- [x] Create the public version of the website
- [x] Create internal pages that render according HCP
- [x] Make sure internal pages are rendered according to affiliation
- [x] Create visualisation components
- [ ] review

### CODE REVIEW

File: apps/molgenis-viz/src/utils/utils.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 // retrive data from a given endpoint using graphql
 //
+// @param url the endpoint to send the request
 // @param query string containing a graphql query
 //
 // @return json
-export async function fetchData(query) {
-  const response = await fetch("graphql", {
+export async function fetchData(url, query) {
+  const response = await fetch(url, {
```
Comment (chinook25) [inline apps/molgenis-viz/src/utils/utils.js:9]: replace with client?

**Review by chinook25 — CHANGES_REQUESTED** [review]
Mostly looks good. Haven't tested yet.

File: apps/cranio-public/README.md
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/cranio-public/README.md:1]: Removed file

File: apps/cranio-public/src/styles/index.scss [hunk 2]
```diff
@@ -0,0 +1,171 @@
+.app-page,
+.page-footer {
+  font-family: Helvetica, sans-serif;
```
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/cranio-public/src/styles/index.scss:3]: That's a good idea. In the next version of the molgenis-viz library, I'm planning on consolidating many of the styles into the main css file, and create more sass mixins.

[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/cranio-public/src/styles/index.scss:3]: For now, I'll leave it in

File: apps/molgenis-viz/src/components/viz/ChartLegend.vue [hunk 3]
```diff
@@ -56,6 +62,12 @@
 export default {
   name: "ChartLegend",
   props: {
+    // a unique ID for the legend
```
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/molgenis-viz/src/components/viz/ChartLegend.vue:65]: The comments are for the vuese doc generator.

File: apps/molgenis-viz/src/components/viz/GroupedColumnChart.vue [hunk 4]
```diff
@@ -0,0 +1,593 @@
...
+  scaleLinear,
+  axisLeft,
+  group,
+  scaleOrdinal,
+  schemePuBuGn,
+};
+
+import ChartLegend from "./ChartLegend.vue";
+import { validateNumRange } from "../../utils/utils.js";
+
```
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/molgenis-viz/src/components/viz/GroupedColumnChart.vue:118]: Same as above. The comments are for the molgenis-viz documentation.

File: apps/molgenis-viz/src/components/viz/GaugeChart.vue [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+    valueFill: {
+      type: String,
+      // `#1c9099`
+      default: "#01665e",
+    },
+
+    // Set the color of the background layer
+    baseFill: {
+      type: String,
+      // default: '#c7eae5'
```
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/molgenis-viz/src/components/viz/GaugeChart.vue:63]: The primary components in the molgenis-viz library have documentation written in the file. The markup is used to generate the user manuals and provide an overview on the default values.

File: apps/cranio-public/src/utils/utils.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export async function postQuery(url, query) {
+  const response = await fetch(url, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
```
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/cranio-public/src/utils/utils.js:5]: Removed!

File: apps/cranio-provider/src/utils/utils.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+export async function postQuery(url, query) {
+  const response = await fetch(url, {
```
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/cranio-provider/src/utils/utils.js:2]: Done!

File: apps/cranio-provider/src/components/AppFooter.vue [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+      <div class="footer-content width-medium">
+        <div class="footer-column footer-links">
+          <p><strong>ERN CRANIO</strong></p>
+          <UnorderedList listType="none">
+            <li><router-link :to="{ name: 'home' }">Home</router-link></li>
+            <li><router-link :to="{ name: 'about' }">About</router-link></li>
+            <li>
+              <router-link :to="{ name: 'dashboard' }">Dashboard</router-link>
+            </li>
+            <!-- more links here -->
```
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/cranio-provider/src/components/AppFooter.vue:13]: Removed.

I plan on adding this component to the `molgenis-viz` library and using named slots. At the time the component was created, it wasn't clear how many projects would use this footer component or the default molgenis footer.

File: apps/cranio-provider/README.md
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/cranio-provider/README.md:1]: Deleted

**Review by chinook25 — APPROVED** [review]

**Pull request merged**